platform/windows/export/template_modifier.cpp

```diff
@@ -0,0 +1,584 @@
+	Vector<uint8_t> resources;
+	resources.resize(sizeof(RESOURCE_DIRECTORY_TABLES));
+	memcpy(resources.ptrw(), RESOURCE_DIRECTORY_TABLES, sizeof(RESOURCE_DIRECTORY_TABLES));
+
+	Vector<Vector<uint8_t>> data_entries;
+	for (Vector<uint8_t> image : p_group_icon.images) {
+		data_entries.append(image);
+	}
+	data_entries.append(p_group_icon.save());
+	data_entries.append(p_version_info.save());
+
+	uint32_t offset = sizeof(RESOURCE_DIRECTORY_TABLES) + data_entries.size() * ResourceDataEntry::SIZE;
+
+	for (Vector<uint8_t> data_entry : data_entries) {
+		ResourceDataEntry resource_data_entry;
+		resource_data_entry.rva = p_virtual_address + offset;
+		resource_data_entry.size = data_entry.size();
+		resources.append_array(resource_data_entry.save());
+		offset += resource_data_entry.size;
+		while (offset % 4) {
+			offset += 1;
+		}
+	}
+
+	for (Vector<uint8_t> data_entry : data_entries) {
+		resources.append_array(data_entry);
+		while (resources.size() % 4) {
+			resources.append(0);
+		}
+	}
+
+	return resources;
+}
+
+TemplateModifier::VersionInfo TemplateModifier::_create_version_info(const HashMap<String, String> &strings) const {
+	StringTable string_table;
+	for (KeyValue<String, String> E : strings) {
+		string_table.put(E.key, E.value);
+	}
+
+	StringFileInfo string_file_info;
+	string_file_info.string_table = string_table;
+
+	FixedFileInfo fixed_file_info;
+	if (strings.has("FileVersion")) {
+		fixed_file_info.set_file_version(strings["FileVersion"]);
+	}
+	if (strings.has("ProductVersion")) {
+		fixed_file_info.set_product_version(strings["ProductVersion"]);
+	}
+
+	VersionInfo version_info;
+	version_info.value = fixed_file_info;
+	version_info.string_file_info = string_file_info;
+
+	return version_info;
+}
+
+String TemplateModifier::_get_icon_path(const Ref<EditorExportPreset> &p_preset, bool p_console_icon) const {
+	String icon_path = ProjectSettings::get_singleton()->globalize_path(p_preset->get("application/icon"));
+	if (p_console_icon) {
+		String console_icon_path = ProjectSettings::get_singleton()->globalize_path(p_preset->get("application/console_wrapper_icon"));
+		if (!console_icon_path.is_empty() && FileAccess::exists(console_icon_path)) {
+			icon_path = console_icon_path;
+		}
+	}
+	return icon_path;
+}
+
+TemplateModifier::GroupIcon TemplateModifier::_create_group_icon(const String &p_icon_path) const {
+	GroupIcon group_icon = GroupIcon();
+
+	Ref<FileAccess> icon_file = FileAccess::open(p_icon_path, FileAccess::READ);
+	if (icon_file.is_null()) {
+		group_icon.fill_with_godot_blue();
+		return group_icon;
+	}
+
+	group_icon.load(icon_file);
+
+	return group_icon;
+}
+
+Error TemplateModifier::_truncate(const String &path, uint32_t size) const {
+	Error error;
+
+	Ref<FileAccess> file = FileAccess::open(path, FileAccess::READ, &error);
+	ERR_FAIL_COND_V(error, ERR_CANT_OPEN);
```

Comment:
```suggestion
	ERR_FAIL_COND_V(error != OK, ERR_CANT_OPEN);
```
Be consistent about error check here (you use `!= OK` below)